src/main/java/com/google/api/generator/engine/ast/StringObjectValue.java

```diff
@@ -37,6 +40,38 @@ public TypeNode type() {
   public abstract static class Builder {
     public abstract Builder setValue(String value);
 
-    public abstract StringObjectValue build();
+    public abstract StringObjectValue autoBuild();
+
+    public StringObjectValue build() {
+      // `\"` is added to the escaped string value for interpreting it correctly in file.
+      String value =
+          String.format("\"%s\"", StringValueEscaper.getInstance().escape(autoBuild().value()));
+      setValue(value);
+      return autoBuild();
+    }
+  }
+
+  private static class StringValueEscaper extends Escaper {
+    private static final Escaper escaper =
+        Escapers.builder()
+            .addEscape('\t', "\\t")
+            .addEscape('\b', "\\b")
+            .addEscape('\n', "\\n")
+            .addEscape('\r', "\\r")
+            .addEscape('\f', "\\f")
+            .addEscape('"', "\\\"")
+            .addEscape('\\', "\\\\")
+            .build();
+
+    private StringValueEscaper() {}
+
+    @Override
+    public String escape(String sourceString) throws EscaperException {
```

Comment:
As this is a RuntimeException, is the `throws` needed? We discussed this last week and in a prior review.

Reply:
yes! it's a runtimeException, while I was thinking that throwing a `EscaperException` can give us more information if our program fails for this. WDYT?